boa/src/builtins/regexp/mod.rs

```diff
@@ -502,4 +502,36 @@ impl RegExp {
 
         Ok(result)
     }
+
+    /// `RegExp.prototype[ @@search ]( string )`
+    ///
+    /// The `[@@search]` method executes a search for a match between a `this` regular expression and a string.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-regexp.prototype-@@search
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@search
+    pub(crate) fn search(this: &Value, arg_str: String, ctx: &mut Context) -> Result<Value> {
```

Comment:
Someone else can confirm, but I'm pretty sure the signature needs to be `(this: &Value, args: &[Value], ctx: &mut Context) -> Result<Value>`, all builin functions have the same signature.
This also needs to be declared as a method of the `RegExp` global object, you can see some examples here: https://github.com/boa-dev/boa/blob/2cb24427f1b5e608b453514af413004d3292c2f8/boa/src/builtins/regexp/mod.rs#L81-L83
Note that after the algorithm definition the spec mentions the `"name"` of the function should is `"[Symbol.search]"`.